core/src/main/java/org/testcontainers/lifecycle/Startables.java

```diff
@@ -72,6 +73,18 @@ public Thread newThread(Runnable r) {
             })
             .toArray(CompletableFuture[]::new);
 
-        return CompletableFuture.allOf(futures);
+        return allOfShortcircuiting(futures);
+    }
+
+    private static CompletableFuture<Void> allOfShortcircuiting(CompletableFuture[] futures) {
```

Comment:
nit: I would name it "firstFailure" or similar

Reply:
Sure, I found this one exceptionally hard to name with something that felt _right_